app/controllers/incomeSources/add/BusinessAddedObligationsController.scala

```diff
@@ -0,0 +1,124 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.incomeSources.add
+
+import auth.MtdItUser
+import config.featureswitch.{FeatureSwitching, IncomeSources}
+import config.{AgentItvcErrorHandler, FrontendAppConfig, ItvcErrorHandler}
+import controllers.agent.predicates.ClientConfirmedController
+import controllers.predicates._
+import models.incomeSourceDetails.BusinessDetailsModel
+import models.incomeSourceDetails.viewmodels.{DatesModel, ObligationsViewModel}
+import models.nextUpdates.{NextUpdateModel, NextUpdatesErrorModel, ObligationsModel}
+import play.api.Logger
+import play.api.i18n.I18nSupport
+import play.api.mvc._
+import services.{DateServiceInterface, IncomeSourceDetailsService, NextUpdatesService}
+import uk.gov.hmrc.auth.core.AuthorisedFunctions
+import views.html.incomeSources.add.BusinessAddedObligations
+
+import java.time.LocalDate
+import java.time.Month.APRIL
+import javax.inject.Inject
+import scala.concurrent.duration.{Duration, MILLISECONDS}
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+class BusinessAddedObligationsController @Inject()(authenticate: AuthenticationPredicate,
+                                                   val authorisedFunctions: AuthorisedFunctions,
+                                                   checkSessionTimeout: SessionTimeoutPredicate,
+                                                   retrieveNino: NinoPredicate,
+                                                   val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                                   val retrieveBtaNavBar: NavBarPredicate,
+                                                   val itvcErrorHandler: ItvcErrorHandler,
+                                                   incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                   val obligationsView: BusinessAddedObligations,
+                                                   nextUpdatesService: NextUpdatesService)
+                                                  (implicit val appConfig: FrontendAppConfig,
+                                                   implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                   implicit override val mcc: MessagesControllerComponents,
+                                                   val ec: ExecutionContext,
+                                                   dateService: DateServiceInterface)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching {
+
+
+  def show(id: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
```

Comment:
ideally we should use incomeSourceId instead of 'id', but this is really optional I would say